src/main/java/gregtech/common/items/behaviors/ItemMagnetBehavior.java

```diff
@@ -96,7 +119,7 @@ public void onUpdate(ItemStack itemStack, Entity entity) {
 
     @SubscribeEvent
     public void onItemToss(@Nonnull ItemTossEvent event) {
-        if (!isActive || event.getPlayer() == null)
+        if (!isActive(event.getEntityItem().getItem()) || event.getPlayer() == null)
```

Comment:
I don't think this is correct. Most of the time the tossed item will not be the magnet, so checking if the tossed item is active will not reflect if the magnet is active